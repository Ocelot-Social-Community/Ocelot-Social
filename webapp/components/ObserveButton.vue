<template>
  <action-button
    :loading="false"
    :disabled="disabled"
    :filled="isObserved"
    :count="count"
    :text="$t('observeButton.observed')"
    icon="bell"
    circle
    @click="toggle"
  />
</template>

<script>
import ActionButton from '~/components/ActionButton.vue'

export default {
  components: {
    ActionButton,
  },
  props: {
    count: { type: Number, default: 0 },
    postId: { type: String, default: null },
    isObserved: { type: Boolean, default: false },
  },
  methods: {
    toggle() {
      this.$emit('toggleObservePost', this.postId, !this.isObserved)
    },
  },
}
</script>
