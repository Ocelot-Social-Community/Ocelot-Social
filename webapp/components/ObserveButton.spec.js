import { mount } from '@vue/test-utils'
import ObserveButton from './ObserveButton.vue'

const localVue = global.localVue

describe('ObserveButton', () => {
  let mocks

  const Wrapper = (count = 1, postId = '123', isObserved = true) => {
    return mount(ObserveButton, {
      mocks,
      localVue,
      propsData: {
        count,
        postId,
        isObserved,
      },
    })
  }

  let wrapper

  beforeEach(() => {
    mocks = {
      $t: jest.fn(),
    }
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders unobserved', () => {
    wrapper = Wrapper(1, '123', false)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits toggleObservePost when clicked', () => {
    wrapper.find('.base-button').trigger('click')
    expect(wrapper.emitted().toggleObservePost).toBeTruthy()
  })
})
