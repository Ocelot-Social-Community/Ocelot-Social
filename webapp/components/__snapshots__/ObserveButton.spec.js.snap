// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObserveButton renders 1`] = `
<div
  class="ds-space text-align-center"
  style="margin-top: 4px; margin-bottom: 4px;"
>
  <button
    class="base-button --icon-only --circle --filled"
    type="button"
  >
    <span
      class="base-icon"
    >
      <!---->
    </span>
     
    <!---->
     
  </button>
   
  <div
    class="ds-space"
    style="margin-bottom: 4px;"
  />
   
  <p
    class="ds-text observe-button-text ds-text-soft"
  >
    <h3
      class="ds-heading ds-heading-h3"
      style="display: inline;"
    >
      1x
    </h3>
    
    
  
  </p>
</div>
`;

exports[`ObserveButton renders unobserved 1`] = `
<div
  class="ds-space text-align-center"
  style="margin-top: 4px; margin-bottom: 4px;"
>
  <button
    class="base-button --icon-only --circle"
    type="button"
  >
    <span
      class="base-icon"
    >
      <!---->
    </span>
     
    <!---->
     
  </button>
   
  <div
    class="ds-space"
    style="margin-bottom: 4px;"
  />
   
  <p
    class="ds-text observe-button-text ds-text-soft"
  >
    <h3
      class="ds-heading ds-heading-h3"
      style="display: inline;"
    >
      1x
    </h3>
    
    
  
  </p>
</div>
`;
